src/reducers/TasksReducer.js

```diff
@@ -70,78 +44,51 @@ function clearChangelog(existingTasks, { task }) {
   })
 }
 
-function currentTimeString() {
-  const date = new Date()
-  return date.toISOString()
-}
-
-function snoozeTasks(existingTasks) {
-  const snoozedTasks = []
+function updateTask(existingTasks, taskToUpdate, newValues) {
   const updatedTasks = existingTasks.map(task => {
-    if (task.isSelected) {
-      const snoozedAt = currentTimeString()
-      snoozedTasks.push(task)
-      return Object.assign({}, task, {
-        snoozedAt,
-        archivedAt: null,
-        isSelected: false,
-        ignore: false,
-      })
+    if (task.storageKey === taskToUpdate.storageKey) {
+      return Object.assign({}, task, newValues)
     }
     return task
   })
   return updatedTasks
 }
 
-function ignoreTasks(existingTasks) {
-  const ignoredTasks = []
-  const updatedTasks = existingTasks.map(task => {
-    if (task.isSelected) {
-      ignoredTasks.push(task)
-      return Object.assign({}, task, {
-        ignore: true,
-        isSelected: false,
-        snoozedAt: null,
-        archivedAt: null,
-      })
-    }
-    return task
-  })
-  return updatedTasks
+function snoozeTasks(existingTasks, { task: taskToUpdate }) {
+  const newValues = {
+    snoozedAt: new Date().toISOString(),
+    archivedAt: null,
+    ignore: false,
+  }
+  return updateTask(existingTasks, taskToUpdate, newValues)
 }
 
-function archiveTasks(existingTasks) {
-  const updatedTasks = existingTasks.map(task => {
-    if (task.isSelected) {
-      const archivedAt = currentTimeString()
-      return Object.assign({}, task, {
-        archivedAt,
-        snoozedAt: null,
-        isSelected: false,
-        ignore: false,
-        changelog: {},
-      })
-    }
-    return task
-  })
-  return updatedTasks
+function ignoreTasks(existingTasks, { task: taskToUpdate }) {
```

Comment:
It seems odd that the method name is plural but it only ever works on a single task.